File: LeetCode75/array_and_string/can_place_flowers/main.py
from typing import List


class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if len(flowerbed) == 1:
            return n <= (1 if flowerbed[0] == 0 else 0)

        i, total = 0, 0
        while i < len(flowerbed):
            if i == 0:
                if flowerbed[i] == 0 and flowerbed[i+1] == 0:
                    i, total = i+1, total+1
            elif i == len(flowerbed) - 1:
                if flowerbed[i] == 0 and flowerbed[i-1] == 0:
                    i, total = i+1, total+1
            else:
                if flowerbed[i] == 0 and flowerbed[i-1] == 0 and flowerbed[i+1] == 0:
                    i, total = i+1, total+1
            i += 1
        return n <= total

File: LeetCode75/array_and_string/can_place_flowers/main.c
#include <stdbool.h>

bool canPlaceFlowers(int* flowerbed, int flowerbedSize, int n)
{
    if (flowerbedSize == 1)
        return ((flowerbed[0] == 0) ? 1 : 0) >= n;

    int total = 0;
    int i = 0;
    while (i < flowerbedSize)
    {
        if (i == 0)
        {
            if (flowerbed[i] == 0 && flowerbed[i + 1] == 0)
            {
                ++total;
                ++i;
            }
        }
        else if (i == flowerbedSize - 1)
        {
            if (flowerbed[i] == 0 && flowerbed[i - 1] == 0)
            {
                ++total;
                ++i;
            }
        }
        else if (flowerbed[i - 1] == 0 && flowerbed[i] == 0
                 && flowerbed[i + 1] == 0)
        {
            ++total;
            ++i;
        }
        ++i;
    }
    return n <= total;
}

#include <stdio.h>

int main(void)
{
    int  flowerbed[] = {1, 0, 0, 0, 1};
    bool result = canPlaceFlowers(flowerbed,
                                  sizeof(flowerbed) / sizeof(flowerbed[0]), 2);
    printf("%d\n", result);
}